### PR TITLE
Rename the MongoSnap fields to JujuDBSnap to match snap name.

### PR DESCRIPTION
**Pull request by howbazaar**
In order to avoid future confusion, we want to have the controller config
named 'juju-db-snap-channel' because the channel is for the juju-db snap.

This also means should we change what the underlying db is, we don't have
to change the config name. Although perhaps that is wishful thinking.

### CODE REVIEW

File: agent/format-2.0.go [hunk 1]
```diff
@@ -60,7 +60,7 @@ type format_2_0Serialization struct {
 	SystemIdentity     string `yaml:"systemidentity,omitempty"`
 	MongoVersion       string `yaml:"mongoversion,omitempty"`
 	MongoMemoryProfile string `yaml:"mongomemoryprofile,omitempty"`
-	MongoSnapChannel   string `yaml:"mongosnapchannel,omitempty"`
+	JujuDBSnapChannel  string `yaml:"juju-db-snap-channel,omitempty"`
```
Comment (wallyworld) [inline agent/format-2.0.go:63]: Perhaps can we make this `jujudb-snap-channel`

The other fields don't have hyphens (they should) and reducing the number we use in this new field looks a little tidier IMO

File: controller/config.go [hunk 2]
```diff
@@ -133,9 +133,9 @@ const (
 	// detault
 	MongoMemoryProfile = "mongo-memory-profile"
 
-	// MongoSnapChannel selects the channel to use when installing mongo
+	// JujuDBSnapChannel selects the channel to use when installing mongo
 	// snaps for focal or later. The value is ignored for older releases.
-	MongoSnapChannel = "mongo-snap-channel"
+	JujuDBSnapChannel = "juju-db-snap-channel"
```
Comment (wallyworld) [inline controller/config.go:138]: and here wrt other suggestion

File: worker/agentconfigupdater/manifold_test.go [hunk 3]
```diff
@@ -139,7 +139,7 @@ func (s *AgentConfigUpdaterSuite) TestCentralHubMissing(c *gc.C) {
 				*result = params.ControllerConfigResult{
 					Config: map[string]interface{}{
 						"mongo-memory-profile": "default",
-						"mongo-snap-channel":   controller.DefaultMongoSnapChannel,
+						"juju-db-snap-channel": controller.DefaultJujuDBSnapChannel,
```
Comment (wallyworld) [inline worker/agentconfigupdater/manifold_test.go:142]: ditto

**Comment by howbazaar** [comment]
$$merge$$

**Comment by achilleasa** [comment]
@howbazaar I actually considered renaming this as you did here but decided against it in the end given that the other mongo options (e.g. the memory-profile one) start with `mongo-`. Should we also change any var/const names in the code to align with this change?

**Pull request merged**